### PR TITLE
User FTE shown on Dashboard

### PR DESCRIPTION
**Pull request by Priyanka-Kaintura**
Targets #2819

### Checklist:

- [x] I have performed a self-review of my own code.

### CODE REVIEW

File: app/Http/Controllers/HomeController.php [hunk 1]
```diff
@@ -24,8 +25,13 @@ public function __construct()
     public function index()
     {
         $unreadBook = (session('disable_book_suggestion')) ? null : Book::getRandomUnreadBook();
+        $name = auth()->user()->name;
+        // FTE = Sum of hours booked by an employee in all the projects/(total number of days in month till today *8)
+        $id = DB::table('users')->where('name', $name)->value('id');
+        $efforts = DB::table('project_team_members_effort')->where('project_team_member_id', $id)->whereMonth('added_on', date('m'))->sum('actual_effort');
+        $FTE = $efforts / (date('d') * 8.00);
```
Comment (GauravGusain98) [inline app/Http/Controllers/HomeController.php:32]: There is already a function defined inside user Model that provides fte for user

File: app/Http/Controllers/HomeController.php [hunk 2]
```diff
@@ -24,8 +25,13 @@ public function __construct()
     public function index()
     {
         $unreadBook = (session('disable_book_suggestion')) ? null : Book::getRandomUnreadBook();
+        $name = auth()->user()->name;
```
Comment (GauravGusain98) [inline app/Http/Controllers/HomeController.php:28]: auth()->user() will give you user object which contains all data i.e id, name, email, etc. related to the user

File: resources/views/home.blade.php [hunk 3]
```diff
@@ -15,9 +15,16 @@
             <p>
                 It looks like you're brand new here and do not have access to any features! Please contact the administrator to grant you required access so that you can get rolling!
             </p>
-        </div>
+        </div>    
     @endif
-
+    <div>
+        @if ($FTE < 1)
+            <span>Your FTE: <b style="color : red;">{{$FTE}}</b> </span>
```
Comment (GauravGusain98) [inline resources/views/home.blade.php:22]: We don't prefer to use style tag in blade file as we use bootstrap classes that are compiled and minified. Please check the variables.scss file to see all the color classes available. You can also check code in other pages to see how we can change the text color

File: app/Http/Controllers/HomeController.php [hunk 4]
```diff
@@ -24,8 +25,13 @@ public function __construct()
...
+        // FTE = Sum of hours booked by an employee in all the projects/(total number of days in month till today *8)
+        $id = DB::table('users')->where('name', $name)->value('id');
+        $efforts = DB::table('project_team_members_effort')->where('project_team_member_id', $id)->whereMonth('added_on', date('m'))->sum('actual_effort');
+        $FTE = $efforts / (date('d') * 8.00);
 
-        return view('home')->with(['book' => $unreadBook]);
+        return view('home')->with(['book' => $unreadBook, 'FTE'=>$FTE]);
```
Comment (GauravGusain98) [inline app/Http/Controllers/HomeController.php:34]: You don't have to pass FTE. You can simply use auth()->user() inside blade file

**Review by GauravGusain98 — CHANGES_REQUESTED** [review]
Please check comment

**Comment by GauravGusain98** [comment]
There should be proper gap below the text `Your FTE: 0`. Currently, the books card is very close to FTE 
image

File: resources/views/home.blade.php [hunk 5]
```diff
@@ -15,9 +15,16 @@
             <p>
                 It looks like you're brand new here and do not have access to any features! Please contact the administrator to grant you required access so that you can get rolling!
             </p>
-        </div>
+        </div>    
     @endif
-
+    <div>
+        @if (($data = auth()->user()->ftes['main']) < 1)
+            <span>Your FTE: <b class = 'text-danger'>{{$data}}</b> </span>
```
Comment (GauravGusain98) [inline resources/views/home.blade.php:22]:
```suggestion
            <span>Your FTE: <b class='text-danger'>{{$data}}</b> </span>
```

**Review by GauravGusain98 — CHANGES_REQUESTED** [review]

File: resources/views/home.blade.php [hunk 6]
```diff
@@ -15,9 +15,16 @@
             <p>
                 It looks like you're brand new here and do not have access to any features! Please contact the administrator to grant you required access so that you can get rolling!
             </p>
-        </div>
+        </div>    
     @endif
-
+    <div>
+        @if (($data = auth()->user()->ftes['main']) < 1)
```
[the remark this answers is not in the document]
Comment (Priyanka-Kaintura) [reply resources/views/home.blade.php:21]: I actually could not find any function in User model which gives FTE, however I saw this implementation in one of the views named 'fte-handler.blade.php' and so tried using it.

File: resources/views/home.blade.php (hunk 5 above)
Comment (Priyanka-Kaintura) [reply resources/views/home.blade.php:22]: Okay sir, I will do that.

File: resources/views/home.blade.php (hunk 6 above)
[the remark this answers is not in the document]
Comment (Priyanka-Kaintura) [reply resources/views/home.blade.php:21]: Sir I checked the code as suggested and the method is present in the following directory: 'Modules/User/Entities/User.php' with name 'getFtesAttribute()'which returns and saves ftes as an attribute of the user, which has two key value pair: one is main or current fte while other is amc. so fte['main] gives the user's current fte. So I  think this isn't a wrong way to get user current fte

[the remark this answers is not in the document]
Comment (GauravGusain98) [reply resources/views/home.blade.php:21]: Ok Let me check.

**Review by GauravGusain98 — APPROVED** [review]
LGTM

File: resources/views/home.blade.php (hunk 6 above)
Comment (GauravGusain98) [inline resources/views/home.blade.php:21]: Why initializing a variable inside if condition?

**Review by GauravGusain98 — CHANGES_REQUESTED** [review]
Please check the comment

File: resources/views/home.blade.php [hunk 7]
```diff
@@ -15,9 +15,16 @@
...
+        </div>    
     @endif
-
+    <div>
+        @if (auth()->user()->ftes['main'] < 1)
+            <span>Your FTE: <b class='text-danger'>{{auth()->user()->ftes['main'] }}</b> </span>
+        @else 
+            <span>Your FTE: <b class='text-success'>{{auth()->user()->ftes['main'] }}</b> </span>
+        @endif
+    </div>
```
Comment (GauravGusain98) [inline resources/views/home.blade.php:26]:
```suggestion
    <div>
        <span>Your FTE: <b class="{{ auth()->user()->ftes['main'] < 1 ? 'text-danger' : 'text-success' }}">{{auth()->user()->ftes['main'] }}</b> </span>
    </div>
```

Comment (GauravGusain98) [reply resources/views/home.blade.php:26]: Also add a proper margin bottom so that the FTE should have proper spacing with below section

**Review by GauravGusain98 — CHANGES_REQUESTED** [review]
Please check comment

**Review by GauravGusain98 — APPROVED** [review]
LGTM

**Pull request merged**